### ISSUE TITLE
`u64` field should be serialized as string in json format

### ISSUE THREAD

**Issue by HarukaMa** [AleoHQ/snarkOS#2045]
## 🐛 Bug Report

The json formatter is storing `u64` fields as a number, and it's silently losing some precision for the data. I'm not sure why it happened during the serialization though - I'd expect rust libraries use `u64` instead of `double` for that.

Currently the only place affected is the nonce of the `PartialSolution`.

## Steps to Reproduce

REST endpoint `testnet3/latest/block` reports inaccurate nonce for coinbase solutions.

## Expected Behavior

The data should either be accurate numbers or strings.

## Your Environment

Assume latest

**Comment by niklaslong** [comment]
Thank you for reporting this issue @HarukaMa. This is puzzling, the `Serializer` implementation work fine with serde's JSON, as tested in snarkVM for `PartialSolution`; and this trait is simply what `warp` uses under the hood in order to handle the data. What are you using to query the rest API? 🤔

**Comment by HarukaMa** [comment]
wow, this might be an invalid issue then. It looks like `jq` is actually the culprit here. The snarkOS code is actually producing good results. That's a little embarrassing.

That said, I've seen another consumer of the rest api incorrectly dealing with those large numbers. Maybe using strings could raise the awareness about that?

<hr>

To clarify, there is another block explorer outta there and it's also losing precision on the `nonce` field. The data in the current form is not safe in the javascript environment, as js uses double for all numbers.

**Comment by niklaslong** [comment]
I see, good to know and agreed; we'll definitely consider the switch and if we don't end up making the change we'll document this better. Thanks again!

**Comment by zvolin** [comment]
yeah, `jq` handles numbers only to about 2^53 which is mantissa size in double precision floats. You could try using [`jql`](https://github.com/yamafaktory/jql) instead of `jq`, it will handle u64 and probably also even u128

**Comment by zvolin** [comment]
please note that `jql` will probably be a lot slower than `jq` when reading from stdin (at least that was true a month ago when I was checking). It prefers files provided as an argument

**Comment by 63669733** [comment]
![image](https://user-images.githubusercontent.com/12948697/201253012-7fb643c3-bb8f-4f76-81ff-e43e911a0d87.png)

Please see the screenshot, how can I solve it if I can't connect?

**Comment by howardwu** [comment]
Check in here, is this issue still relevant?

**Comment by HarukaMa** [comment]
Yes, the nonce is still returned in json as a number.

**Comment by raychu86** [comment]
Is this issue still relevant after https://github.com/AleoNet/snarkVM/pull/2559?

**Comment by vicsn** [comment]
This was an issue on the `jq` or `js` side, we are not serializing u64 as string unless someone brings up a strong usecase.
